pkg/codegen/internal/test/sdk_driver.go

```diff
@@ -112,37 +161,86 @@ var sdkTests = []sdkTest{
 // The schema is the only piece that must be manually authored. Once the schema has been written, the expected outputs
 // can be generated by running `PULUMI_ACCEPT=true go test ./..." from the `pkg/codegen` directory.
 func TestSDKCodegen(t *testing.T, language string, genPackage GenPkgSignature) {
-	testDir := filepath.Join("..", "internal", "test", "testdata")
+	TestSDKCodegenWithOptions(t, &TestSDKCodegenOptions{
+		Language:   language,
+		GenPackage: genPackage,
+	})
+}
 
-	for _, tt := range sdkTests {
+// A variant of TestSDKCodegen with more options.
+func TestSDKCodegenWithOptions(t *testing.T, rawOptions *TestSDKCodegenOptions) {
```

Comment:
Perhaps a bit overkill to parameterize by SdkTest[] set, I need to look if I can shoehorn output-funcs test into a normal SdkTest in the default list.